src/core/pointcloud/qgspointclouddataprovider.cpp

```diff
@@ -291,6 +292,26 @@ struct MapIndexedPointCloudNode
         pointAttr[ QStringLiteral( "X" ) ] = x;
         pointAttr[ QStringLiteral( "Y" ) ] = y;
         pointAttr[ QStringLiteral( "Z" ) ] = z;
+
+        const QDateTime gpsBaseTime = QDateTime::fromSecsSinceEpoch( 315964809, Qt::UTC );
```

Comment:
Can you bench test this change? My suspicion is that all these datetime calculations will dramatically slow point cloud iteration. (They can be very slow)

Reply:
Do you mean the point cloud iteration during the `identify()` procedure?

I benchmarked this function which is only called while using the identify tool.
Dataset: [Millsite](https://s3.amazonaws.com/data.entwine.io/millsite.copc.laz)
Procedure: I identified 50 points (areas with points) for both versions, started from the overview of the whole point cloud and zoomed in every 10 points. 

|Function| Before change | After change|
| --- | --- | --- |
|Average | 47,4313725490196 ms | 51,843137254902 ms |
| Median | 48 ms | 45 ms |
| Average worst 5% | 104,6 ms | 131 ms |
| Average best 5% | 12,6 ms | 12,8 ms |